packages/slate-react/src/components/editor.js

```diff
@@ -260,12 +237,11 @@ class Editor extends React.Component {
 
   onChange = change => {
     debug('onChange', { change })
-
-    this.stack.run('onChange', change, this)
-    const { value } = change
+    if (this.value === change.value) return
+    const value = this.processValueOnChange(change.value, this.stack)
     const { onChange } = this.props
-    if (value == this.value) return
-    onChange(change)
+    this.tmp.value = value
```

Comment:
Why this?

Reply:
All our values are updated by `this.props.onChange` and `stack.run('onChange')`.  During the mount, we pass the `stack.run('onChange', this.props.value)` back to the onChange, and the the parent container feed use the value after onChange.  

But we do not want to have onChange called again on the value that we returned to the parent container and then passed back to me.  Because calling onChange again and again on the same value will dangerously lead to infinite loop. 

Therefore I use a tmp.value and tell the get value() not to run onChange if it is already processed by onChange(https://github.com/ianstormtaylor/slate/pull/1975/files#diff-0f1805ffc056f7c54e6a6f98ba0259a1R174)